avm/res/azure-stack-hci/network-interface/main.bicep

```
@@ -0,0 +1,142 @@
+metadata name = 'Azure Stack HCI Network Interface'
+metadata description = 'This module deploys an Azure Stack HCI network interface.'
+
+@description('Required. Name of the resource to create.')
+param name string
+
+@description('Optional. Location for all Resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Tags of the resource.')
+param tags object?
+
+@description('Required. Resource ID of the associated custom location.')
+param customLocationId string
```

```suggestion
param customLocationResourceId string
```
In alignment with spec [SNFR22](https://azure.github.io/Azure-Verified-Modules/specs/bcp/res/#:~:text=ID%3A%20SNFR22%20%2D%20Category%3A%20Inputs%20%2D%20Parameters/Variables%20for%20Resource%20IDs)